tensorflow/python/ops/image_ops.py

```diff
@@ -478,40 +492,46 @@ def crop_to_bounding_box(image, offset_height, offset_width, target_height,
                   the input.
     target_height: Height of the result.
     target_width: Width of the result.
+    dynamic_shape: Whether the input image has undertermined shape. If set to
+      `True`, shape information will be retrieved at run time. Default to
+      `False`.
 
   Returns:
     3-D tensor of image with shape `[target_height, target_width, channels]`
 
   Raises:
     ValueError: If the shape of `image` is incompatible with the `offset_*` or
-    `target_*` arguments
+    `target_*` arguments, and `dynamic_shape` is set to `False`.
   """
   image = ops.convert_to_tensor(image, name='image')
-  _Check3DImage(image, require_static=True)
-  height, width, _ = _ImageDimensions(image)
+  _Check3DImage(image, require_static=(not dynamic_shape))
```

Comment:
These not... are ugly. Can you change the argument from dynamic_shape=False to static_shape_check=True? Then all the functions have the same kind of parameter.